### DISCUSSION TITLE
Verify commit signed with sigstore

### DISCUSSION THREAD

**Discussion by ctron** [community/community#37703] [Code Security]
I would like to use [sigstore](https://www.sigstore.dev/) in a repository. That works with git just fine, but pushing signed commits to GitHub now shows those commits in orange and as "unverified":

![image](https://user-images.githubusercontent.com/202474/198823474-5c608d8e-c041-4dea-92f5-7ab0382016de.png)

So adding things that makes the code actually more secure, gets visually downgraded on GitHub. And it doesn't seem to look like that I can configure this.

I guess what I am looking for is proper sigstore support, or maybe a way to ignore sigstore style signatures if they aren't supported.

**Comment by jfullstackdev** [comment]
seems interesting, it's hosted on GitHub and there are developers you can see

https://github.com/sigstore

you can try asking in the most relevant repo that you will see

**Reply by ctron** [reply]
I am not sure anyone outside of GitHub can do much there. As the git verification is part of GitHub, or the GitHub UI.

**Reply by jfullstackdev** [reply]
as you can see here

https://docs.github.com/en/authentication/managing-commit-signature-verification

they are the official ways to verify commits, that's why I'm pointing you to that project, maybe they are still not recognized by GitHub ?

**Reply by ctron** [reply]
> they are the official ways to verify commits, that's why I'm pointing you to that project, maybe they are still not recognized by GitHub ?

I guess so, but I think this is something that GitHub would need to fix. I am not sure the project could do that for them?

Aside, I still think that the GitHub UI doesn't do a good here either. If GitHub doesn't trust those signatures, then it might be necessary to add an option for adding a trust anchor (so that it can get verified) or to allow disabling the badge (if GitHub is unable to verify the signature, but others might).

**Reply by christophe-kamphaus-jemmic** [reply]
The sigstore project actually has an issue which is just for communicating on Github progress for supporting sigstore commit signatures: https://github.com/sigstore/gitsign/issues/40

Gitlab has an issue for sigstore integration: https://gitlab.com/gitlab-org/gitlab/-/issues/364428

**Reply by ctron** [reply]
So I guess this is the "answer".

**Reply by wlynch** [reply]
Gitsign maintainer here - we keep https://github.com/sigstore/gitsign/issues/40 open because we get this question a lot. As you mentioned, it is unfortunately not something that the Sigstore project has much control of, since it comes down to what roots GitHub trusts to show the green checkmark. 

Currently [GitHub uses the Debian ca-certificates package](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#smime-commit-signature-verification), which uses the [Mozilla CA List](https://wiki.mozilla.org/CA/Included_Certificates), which Sigstore is not a part of. We've had some discussions internally of what it would take to get Sigstore added to the list, but since it is primarily targeted towards web browser certificates some of the requirements don't necessarily make sense for Sigstore (particularly around revocation requirements since Fulcio only issues short-lived code signing certs).

I'd love to keep this discussion open as a signal on the GitHub side for adding support for the Sigstore root. No one wants to see this happen more than me. 🙂

**Reply by ctron** [reply]
@wlynch Thanks for letting us know. I unmarked the "answer", keeping it open as you suggested. Which I think makes sense.
